### ISSUE TITLE
Issue with ws_distance

### ISSUE THREAD

**Issue by giovannipizzi** [wannier-developers/wannier90#210]
We (@AntimoMarrazzo and me) have a problem with a specific input.

The code prints `ERROR in hamiltonian_wigner_seitz: error in finding Wigner-Seitz points`.

There are two issues here:
1. following the code path, I think the routine is always called, also when `use_ws_distance` is False. Maybe we should skip that part, so that if an error occurs, at least the code does not stop when setting `use_ws_distance=.false.`.
2. I don't know why the code stops. I attach here the input [tmp.win](https://github.com/wannier-developers/wannier90/files/2537934/tmp.win.txt) and the output with high-verbosity 
[tmp.wout](https://github.com/wannier-developers/wannier90/files/2537935/tmp.wout.txt) (rename the files removing `.txt` if you want to run again). There are many kpoints so it is hard to check "by eye".
(it finds `tot = 3524.00000000000`, while the number of points in the `mp_grid` is `3528`).
We even tried to increase from a 2x2x2 supercell to a larger one and to change a bit one of the thresholds, but apparently this didn't work in the sense that tot is always equal to 3524 (but we should re-check to be sure).

Note: to reproduce without running the DFT calculation, you can just do the following changes:
- in `hamiltonian.F90`, add `public :: hamiltonian_wigner_seitz` at the top and declare `ham_k` as public: `complex(kind=dp), save, allocatable, public :: ham_k(:, :, :)`
- in `wannier_prog.F90`:
  - add `integer :: ierr` at the top
  - add the following block inside `if (postproc_setup)`:
    ```fortran
    print *, '### DEBUG ###'
    if (on_root) call hamiltonian_wigner_seitz(count_pts=.true.)
    allocate (irvec(3, nrpts), stat=ierr)
    if (ierr /= 0) call io_error('Error in allocating irvec in hamiltonian_setup')
    irvec = 0
    !
    allocate (ndegen(nrpts), stat=ierr)
    if (ierr /= 0) call io_error('Error in allocating ndegen in hamiltonian_setup')
    ndegen = 0
    !
    allocate (ham_r(num_wann, num_wann, nrpts), stat=ierr)
    if (ierr /= 0) call io_error('Error in allocating ham_r in hamiltonian_setup')
    ham_r = cmplx_0
    !
    allocate (ham_k(num_wann, num_wann, num_kpts), stat=ierr)
    if (ierr /= 0) call io_error('Error in allocating ham_k in hamiltonian_setup')
    ham_k = cmplx_0
    print *, '### first done ###'
    if (on_root) call hamiltonian_wigner_seitz(count_pts=.false.)
    ```
    and then run `wannier90 -pp tmp`.

@paulatz maybe you could help?

**Comment by gibertini** [comment]
The issue seems to arise from the fact that when unit cells are very anisotropic (e.g. very stretched in one direction, as in this case) the original range of points among which the code searches for points inside the Wigner-Seitz cell was insufficient.

The range is now larger to comply with these cases and it's no longer hard coded, so that in case of need it can be further enlarged by changing the variable search_size in hamiltonian_wigner_seitz (see #216).

Concerning the problem that this issue arises also when use_ws_distance=.false.: this is due to the fact that hamiltonian_wigner_seitz is called always at the beginning to initialise the variable nrpts and thus allocate a few arrays. 
I think this can be avoided, and I will work on it.

**Comment by giovannipizzi** [comment]
Fixed in #216
